main/src/ca/uwaterloo/flix/api/lsp/provider/ImplementationProvider.scala

```diff
@@ -0,0 +1,23 @@
+package ca.uwaterloo.flix.api.lsp.provider
+
+import ca.uwaterloo.flix.api.lsp.{LocationLink, Position}
+import ca.uwaterloo.flix.language.ast.TypedAst.Root
+
+object ImplementationProvider {
+
+  /**
+    * Returns implementations LocationLink for a given class.
+    */
+  def processImplementation(uri: String, position: Position)(implicit root: Root): List[LocationLink] = {
+    if (root == null) {
+      // No AST available.
+      return Nil
+    }
+
+    root.instances.keys.filter(c => c.loc.source.name == uri
+      && c.loc.beginLine == position.line
+      && c.loc.beginCol <= position.character
+      && (c.loc.endLine != position.line || c.loc.endCol >= position.character)
```

Comment:
Whats with all this complicated logic here? Whats the intuition and why is it needed?

Reply:
So the logic here is that we want to intercept the symbol pointed by `position`. In order to achieve this we need check that the position is actually inside our instance.

Reply:
Actually I revisited it since it is not really right, the logic is:
- the line should be in the interval [beginLine, endLine]
- the character should be in the greater than beginCol if line == beginLine (otherwise it can be also less than)
- the same condition (in reverse) applies to the character with the endCol boundary.